### PR TITLE
Private Asks

### PR DESCRIPTION
**Pull request by bjfresh**
Functional:

- createAsk()
- cancelAsk()
- fillAsk()
- setAskPrice()

### CODE REVIEW

File: @market/hooks/usePrivateAskTransaction.ts [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+  collectionAddress?: string
+  tokenId?: string
+  onNext?: () => void
+}
+
+export const usePrivateAskTransaction = ({
+  nft: nftData,
+  onNext,
+}: usePrivateAskTransactionProps) => {
+  const { PrivateAsks } = useContractContext() // Should this all be moved to usePrivateAskContext?
```
Comment (olegakbarov) [inline @market/hooks/usePrivateAskTransaction.ts:31]: worth having a conversation about this stuff, different approaches have different trade-offs

File: @shared/components/ExchangeValue/ExchangeValueAmount.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  amount?: string
+  options?: any
+}
+
+export function ExchangeValueAmount({
+  amount,
+  currencyAddress = AddressZero,
+  children,
+  ...props
+}: ExchangeValueAmountProps) {
```
Comment (olegakbarov) [inline @shared/components/ExchangeValue/ExchangeValueAmount.tsx:23]: this does not look right :)

File: @shared/utils/validateCurrency.ts [hunk 3]
```diff
@@ -1,14 +1,48 @@
 import { BigNumber } from '@ethersproject/bignumber'
+import { parseUnits } from '@ethersproject/units'
+import { isNaN } from 'lodash'
 
-export const validateCurrency = ({ amount }: { amount: string | number }) => {
-  if (amount === 0) {
+// export const validateCurrency = ({ amount }: { amount: string | number }) => {
```
Comment (olegakbarov) [inline @shared/utils/validateCurrency.ts:5]: probably worth removing

File: @shared/components/ExchangeValue/ExchangeValueAmount.tsx (hunk 2 above)
Comment (bjfresh) [reply @shared/components/ExchangeValue/ExchangeValueAmount.tsx:23]: Removing the file entirely for now :)

**Review by olegakbarov — APPROVED** [review]

**Pull request merged**